cmd/api/main.go

```diff
@@ -180,19 +180,19 @@ func main() {
 			recovery.StreamServerInterceptor(recovery.WithRecoveryHandler(recoveryHandler)),
 		),
 	)
-	v1alpha2pb.RegisterResultsServer(gs, v1a2)
+	v1alpha3pb.RegisterResultsServer(gs, v1a2)
```

Comment:
I did the tests, but I think something is getting wrong. If I update the server with the new code on the same DB, the logs are not being stored. I will try to find out the reason and fix.

Reply:
We change the API version for breaking change only. Otherwise, you need to add proto for both v1alpha2 and v1alpha3.  Adding a field or method isn't a breaking change. Renaming or removing them is a breaking change.

In this PR, I don't find any new field or method in proto. Only an internal behaviour change which I don't think requires a breaking change.

Reply:
Should we roll back to just migrating the DB?

Reply:
Why do we need to migrate the DB? Isn't it part of json field?

Reply:
Yes but existing data will not have this field. Although the condition is handled in getLog logic. If that is not a concern then we may not bump the API version. 

But then how do we indicate the change in Log object? Or is it not needed?

Reply:
Why don't we update the Log Object version? Why is it necessary for it to have the same version as API?

Reply:
It is not, we discussed in the call that to keep things in sync we can update the whole API. Do you think we should rediscuss the context and implementation for this in the next call?

Reply:
Yes, we should rediscuss this.

Reply:
adding object specific versioning is atypical at least from a k8s perspective

is there precedent in tekton @khrm that I'm missing?

that said @avinal if you and @sayan-biswas have not had a chance to get together and sort out the issues in your testing that we discussed, at this point I'm up for any path with is both a) not too crazy, and b) does not require a migrator

Reply:
We are updating the Object version here as well as the API. That object is internal to Results. That was the context of my comment. Let's discuss this in WG call.